walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -183,4 +184,28 @@ public static TimeRange fromStartEnd(int start, int end, boolean inclusive) {
   public static TimeRange fromStartDuration(int start, int duration) {
     return new TimeRange(start, duration);
   }
+
+  /**
+   * Uses a branchless binary search to find the first element in a TimeRange @param collection
```

Comment:
Please add to the documentation what "potentially overlaps" means and how the method behaves in boundary cases. Comments inside the method explaining how your math works, in natural langauge, would improve reading speed, too.